### ISSUE TITLE
setting fields during import not working

### ISSUE THREAD

**Issue by adamjakab** [beetbox/beets#3487]
I am unable to set any fields during import. I would like to do this so that after import I can distinguish the newly imported files from the rest of the library. 

### Problem

Running the import command without any flags (so set_fields config options should be applied from the configuration file [below]) imports the files, but:

```sh
$ beet import /Volumes/Data/Download/Music/2Pac/
$ beet ls artist:2Pac -f '$title - $encoder - $grouping - $collection - $comments'
```
produces the following output:
```
2 Of Amerikaz Most Wanted -  -  - $collection - 
All About U -  -  - $collection - 
Brenda's Got A Baby -  -  - $collection - 
California Love (Original Version) -  -  - $collection - 
Changes -  -  - $collection - 
Dear Mama -  -  - $collection - 
God Bless The Dead -  -  - $collection - 
Hail Mary -  -  - $collection - 
Heartz Of Men -  -  - $collection - 
Hit 'Em Up -  -  - $collection - 
How Do U Want It -  -  - $collection - 
How Long Will They Mourn Me -  -  - $collection - 
I Ain't Mad At Cha -  -  - $collection - 
I Get Around -  -  - $collection - 
Keep Ya Head Up -  -  - $collection - 
Life Goes On -  -  - $collection - 
Me Against The World -  -  - $collection - 
Picture Me Rollin' -  -  - $collection - 
So Many Tears -  -  - $collection - 
Temptations -  -  - $collection - 
To Live & Die In L.A- -  -  - $collection - 
Toss It Up -  -  - $collection - 
Trapped -  -  - $collection - 
Troublesome 96' -  -  - $collection - 
Unconditional Love -  -  - $collection - 
```
Neither of the fields has the value assigned. 
Moreover `collection` field is not recognised. (which makes sense because it is not listed by `beet fields` but I wanted to mention it because it is suggested by the [docs](https://beets.readthedocs.io/en/stable/reference/config.html#id91).

If I try to set any of the same fields manually during import:
```
$ beet import --set encoder="jack" --set grouping="newstuff" /Volumes/Data/Download/Music/2Pac/
```
I get the same results. Fields are not being set during import.
I also tried to disable all plugins but no avail.

However, if I do this and confirm the changes:
```
$ beet mod artist:2Pac encoder=jack
Modifying 25 items.
2Pac - Greatest Hits - 2 Of Amerikaz Most Wanted
  encoder:  -> jack
2Pac - Greatest Hits - All About U
  encoder:  -> jack
2Pac - Greatest Hits - Brenda's Got A Baby
  encoder:  -> jack
2Pac - Greatest Hits - California Love (Original Version)
  encoder:  -> jack
2Pac - Greatest Hits - Changes
  encoder:  -> jack
2Pac - Greatest Hits - Dear Mama
  encoder:  -> jack
2Pac - Greatest Hits - God Bless The Dead
  encoder:  -> jack
2Pac - Greatest Hits - Hail Mary
  encoder:  -> jack
2Pac - Greatest Hits - Heartz Of Men
  encoder:  -> jack
2Pac - Greatest Hits - Hit 'Em Up
  encoder:  -> jack
2Pac - Greatest Hits - How Do U Want It
  encoder:  -> jack
2Pac - Greatest Hits - How Long Will They Mourn Me
  encoder:  -> jack
2Pac - Greatest Hits - I Ain't Mad At Cha
  encoder:  -> jack
2Pac - Greatest Hits - I Get Around
  encoder:  -> jack
2Pac - Greatest Hits - Keep Ya Head Up
  encoder:  -> jack
2Pac - Greatest Hits - Life Goes On
  encoder:  -> jack
2Pac - Greatest Hits - Me Against The World
  encoder:  -> jack
2Pac - Greatest Hits - Picture Me Rollin'
  encoder:  -> jack
2Pac - Greatest Hits - So Many Tears
  encoder:  -> jack
2Pac - Greatest Hits - Temptations
  encoder:  -> jack
2Pac - Greatest Hits - To Live & Die In L.A-
  encoder:  -> jack
2Pac - Greatest Hits - Toss It Up
  encoder:  -> jack
2Pac - Greatest Hits - Trapped
  encoder:  -> jack
2Pac - Greatest Hits - Troublesome 96'
  encoder:  -> jack
2Pac - Greatest Hits - Unconditional Love
  encoder:  -> jack
Really modify and move? (Yes/no/select) yes

$ beet ls artist:2Pac -f '$title - $encoder - $grouping'
2 Of Amerikaz Most Wanted - jack - 
All About U - jack - 
Brenda's Got A Baby - jack - 
California Love (Original Version) - jack - 
Changes - jack - 
Dear Mama - jack - 
God Bless The Dead - jack - 
Hail Mary - jack - 
Heartz Of Men - jack - 
Hit 'Em Up - jack - 
How Do U Want It - jack - 
How Long Will They Mourn Me - jack - 
I Ain't Mad At Cha - jack - 
I Get Around - jack - 
Keep Ya Head Up - jack - 
Life Goes On - jack - 
Me Against The World - jack - 
Picture Me Rollin' - jack - 
So Many Tears - jack - 
Temptations - jack - 
To Live & Die In L.A- - jack - 
Toss It Up - jack - 
Trapped - jack - 
Troublesome 96' - jack - 
Unconditional Love - jack - 
```

So it happens only during import. Any ideas? Am I doing something wrong?

### Setup

* OS: MacOS 10.13
* Python version: 3.7.6
* beets version: 1.4.9
* Turning off plugins made problem go away (yes/no): nope

My configuration (output of `beet config`) is:

```yaml
directory: /Volumes/J/Music/
library: ~/Documents/Assets/Beets_music_library.db
asciify_paths: yes

ui:
    color: yes

plugins: [fromfilename, the, inline, duplicates, badfiles, convert]

import: 
    autotag: no
    write: no
    #move: yes
    copy: yes
    resume: no
    from_scratch: no
    none_rec_action: asis
    duplicate_action: remove
    set_fields:
        encoder: jack
        grouping: newstuff
        collection: unordered
        comments: newimport

musicbrainz:
    host: musicbrainz.org
    ratelimit: 1
    ratelimit_interval: 1.0
    searchlimit: 3 

match:
    strong_rec_thresh: 0.25
    max_rec:
        missing_tracks: none
        unmatched_tracks: medium

paths:
    default: $initial_letter/%the{$albumartist_checked}/$album_checked/$title
    
item_fields:
    albumartist_checked: albumartist if albumartist else 'Unknown'
    album_checked: album if album else 'Unknown'
    initial_letter: albumartist[0].upper() if albumartist else 'U'

badfiles:
    commands:
        mp3: mp3val
        flac: flac --test --warnings-as-errors --silent

convert:
    command: ffmpeg -i $source -y -vn -q:a 5 -f mp3 $dest
    extension: mp3
    max_bitrate: 156
    threads: 1
    auto: false
    dest: /Volumes/J/Music/000_IMPORT/Converted
    format: mp3
    formats:
        mp3: ffmpeg -i $source -y -vn -q:a 5 -f mp3 $dest
        flac: ffmpeg -i $source -y -vn -acodec flac -q:a 5 -f mp3 $dest
```

**Comment by sampsyo** [comment]
Hi! The tricky thing here is that, because you're doing an album-level import, this is setting values only on the albums, not the tracks. Can you try that template with a `beet ls -a`?

This is admittedly confusing and we should probably fix it. I could have sworn there was a GitHub issue or a Discourse thread about doing exactly that… any chance you might have better luck searching for it than I did?

**Comment by adamjakab** [comment]
Hi @sampsyo, thanks for the reply. In the meanwhile i looked into the code and I figured out how the importing and the setting of fields works in the two different 'modes': album and singleton.

In fact `beet import -s --set encoder="jack"` using the singleton mode and setting a track field works perfectly. The same applies to `beet import --set albumstatus="new"` when setting an album field during an album-level import. So, there is no bug here ;)

I guess, the documentation should just mention this distinction briefly.
I did try to look for the issue (quickly) before opening this one but I could not find anything. I will try again.
